### ISSUE TITLE
Adding video overlay feature to the CameraVideoRecorder plugin

### ISSUE THREAD

**Issue by addy1997** [ignitionrobotics/ign-gazebo#785]
## Desired behavior
In the [CameraVideoRecorder](https://github.com/ignitionrobotics/ign-gazebo/blob/ign-gazebo5/src/systems/camera_video_recorder/CameraVideoRecorder.hh) plugin, video is recorded using this service https://github.com/ignitionrobotics/ign-gazebo/blob/51befb46a0159c353aec59a2060df55d3f85e48c/src/systems/camera_video_recorder/CameraVideoRecorder.cc#L83 which is quite slow for our application of recording videos of animals with drones in the same environment.

Therefore, I wanted to add this feature called [video overlay](https://en.wikipedia.org/wiki/Video_overlay) to the plugin to make the process of video recording faster.

## Alternatives considered
We tried implementing a plugin using OpenCV and CameraSensor to implement the same but the library lacks some features. Displaying the image taken by the camera using OpenCV had some issues like these [wobbling issue](https://drive.google.com/file/d/16N6x3quAylhq7LR9K-u2Gz07UXXX3xjv/view?usp=sharing)

## Implementation suggestion
This is based on my understanding of the software and may not be the exact suggestion. If we can include a `node `that `bypasses the physics engines` and directly `displays the video with another window`to make the process faster.

In short, we can have two windows, the regular` gazebo GUI` and another `window for displaying the video`. Something like this

![image](https://user-images.githubusercontent.com/29406906/115983092-d7ecff00-a5bc-11eb-8a4b-ccb9590c3bb3.png)

Firstly, is this possible? If yes, any suggestion is appreciated.

Thanks

**Comment by chapulina** [comment]
> quite slow for our application

You mean that the process of recording the video is slow, or the recorded video shows a slow simulation?

Are you using the `use_sim_time` and / or `lockstep` options explained on the [Video recorded tutorial](https://ignitionrobotics.org/api/gazebo/5.0/videorecorder.html)?

**Comment by addy1997** [comment]
> > quite slow for our application
> 
> You mean that the process of recording the video is slow, or the recorded video shows a slow simulation?
> 
> Are you using the `use_sim_time` and / or `lockstep` options explained on the [Video recorder tutorial](https://ignitionrobotics.org/api/gazebo/5.0/videorecorder.html)?

Yes Ma'am, I have referred to both the tutorials, even this [one](https://ignitionrobotics.org/api/common/3.10/hw-encoding.html). So, I have tried both the options, i.e., recording the video in `use_sim_time ` and `lockstep` on two different machines.

I reckon there is some problem with `identifying the encoder`, not sure though. Well, it shouldn't be coz our setup, RAM, workstaions `(NVIDIA RTX A6000 GPU)` are working fine.

**Comment by addy1997** [comment]
The process of recording the video is slow. I am using both  ```use_sim_time``` and ```lockstep```. I referred to [this ](https://ignitionrobotics.org/api/common/3.10/hw-encoding.html) tutorial as well but couldn't get rid of the errors.
